src/Xamarin.Android.Build.Tasks/Tasks/Aapt.cs

```diff
@@ -388,17 +388,28 @@ protected void LogEventsFromTextOutput (string singleLine, MessageImportance mes
 				if (message.StartsWith ("error: ", StringComparison.InvariantCultureIgnoreCase))
 					message = message.Substring ("error: ".Length);
 
-				LogError ("APT0000", message, file, line);
-				return;
+				if (level.Contains ("error") || (line != 0 && !string.IsNullOrEmpty (file))) {
```

Comment:
Once concern with this change is that *previously*, any message which wasn't a `warning` or `fakeLogOpen` was treated as an error.

Now, *only* messages which contain `error` are errors.

This *sounds* reasonable, but will there be any unexpected fallout? Will we "miss" any errors with this change?

Additionally, it looks like we won't log `message` *at all* if it's not explicitly a `warning` or an `error`. Is *that* desirable? (Presumably we should *always* log, no?)

Reply:
As stated google is not consistent with its labelling of errors/warnings. I went through the `aapt` code and added any additional ones to https://github.com/xamarin/xamarin-android/pull/1047/files#diff-97e1ab3c8027acfecf85b8bb5a88dd9aR409

Reply:
Should we always log the message? Or are there messages we should skip entirely?

It looks like the new version will only log messages which contain `warning`, `fakeLogOpen`, `error`, or contain a filename. Is that desirable?

Reply:
We fall through to https://github.com/dellis1972/xamarin-android/blob/79e432f32f08f4b2c867e07cdd225a5de2c9449a/src/Xamarin.Android.Build.Tasks/Tasks/Aapt.cs#L405 if the regex does not match, or its not one of the `KnownInvalidErrorMessages`

That said, I'm sure we had an old bug where if we do log the error/warning using `LogMessage` as well as `LogError` or `LogWarning` we ended up with the same error appearing in the error list twice (sometimes). I think because msbuild does some parsing too.

Reply:
Looking at the aapt source google are not consistent with labelling.. but :) they are consistent with writing ALL errors/warnings to stderr.. so I will change this PR to log anything we get from stdout as a message. and only process the errors/warnings from stderr.

https://android.googlesource.com/platform/frameworks/base/+/c761d8b0086f9e92eb1e4b3a5683ae6afb84036a/tools/aapt/Resource.cpp#1025

https://android.googlesource.com/platform/frameworks/base/+/c761d8b0086f9e92eb1e4b3a5683ae6afb84036a/tools/aapt/Resource.cpp#218

https://android.googlesource.com/platform/frameworks/base/+/c761d8b0086f9e92eb1e4b3a5683ae6afb84036a/tools/aapt/Resource.cpp#231

https://android.googlesource.com/platform/frameworks/base/+/c761d8b0086f9e92eb1e4b3a5683ae6afb84036a/tools/aapt/Resource.cpp#197